prover_rust/src/prover.rs

```diff
@@ -0,0 +1,264 @@
+            status: ProofStatus::Error,
+            ..Default::default()
+        };
+        match task.task_type {
+            ProofType::ProofTypeBatch => {
+                let chunk_hashes_proofs = self.gen_chunk_hashes_proofs(task)?;
+                let batch_proof = handler.aggregator_gen_agg_evm_proof(chunk_hashes_proofs,
+                    None,
+                    self.get_output_dir())?;
+
+                proof_detail.batch_proof = Some(batch_proof);
+                Ok(proof_detail)
+            },
+            ProofType::ProofTypeChunk => {
+                let chunk_trace = self.gen_chunk_traces(task)?;
+                let chunk_proof = handler.prover_gen_chunk_proof(chunk_trace,
+                    None,
+                    None,
+                    self.get_output_dir())?;
+
+                proof_detail.chunk_proof = Some(chunk_proof);
+                Ok(proof_detail)
+            },
+            _ => bail!("task type invalid")
+        }
+    }
+
+    pub fn submit_proof(&self, proof_detail: &ProofDetail, uuid: String) -> Result<()> {
+        let proof_data = match proof_detail.proof_type {
+            ProofType::ProofTypeBatch => {
+                serde_json::to_string(proof_detail.batch_proof.as_ref().unwrap())?
+            },
+            ProofType::ProofTypeChunk => {
+                serde_json::to_string(proof_detail.chunk_proof.as_ref().unwrap())?
+            },
+            _ => unreachable!()
+        };
+
+        let request = SubmitProofRequest {
+            uuid,
+            task_id: proof_detail.id.clone(),
+            task_type: proof_detail.proof_type,
+            status: proof_detail.status,
+            proof: proof_data,
+            ..Default::default()
+        };
+
+        self.do_submit(&request)
+    }
+
+    pub fn submit_error(&self, task: &Task, failure_type: ProofFailureType, error: Error) -> Result<()> {
+        let request = SubmitProofRequest {
+            uuid: task.uuid.clone(),
+            task_id: task.id.clone(),
+            task_type: task.task_type,
+            status: ProofStatus::Error,
+            failure_type: Some(failure_type),
+            failure_msg: Some(error.to_string()),
+            ..Default::default()
+        };
+        self.do_submit(&request)
+    }
+
+    fn do_submit(&self, request: &SubmitProofRequest) -> Result<()> {
+        self.coordinator_client.borrow_mut().submit_proof(request)?;
+        Ok(())
+    }
+
+    fn get_configured_block_number_value(&self) -> Result<Option<U64>> {
+        self.get_block_number_value(&self.config.l2geth.confirmations)
+    }
+
+    fn get_block_number_value(&self, block_number: &BlockNumber) -> Result<Option<U64>> {
+        match block_number {
+            BlockNumber::Safe | BlockNumber::Finalized => {
+                let header = self.geth_client.borrow_mut().header_by_number(block_number)?;
+                Ok(header.get_number())
+            },
+            BlockNumber::Latest => {
```

Comment:
only using this path is OK. can also simplify this function, or just call
```
let number = self.geth_client.borrow_mut().block_number()?;
```
outside.

Reply:
this is outdated, the problem is fixed